dataplane/forwarding/fwdaction/actions/select_action_list.go

```diff
@@ -68,29 +71,36 @@ func (s *selectActionList) Process(packet fwdpacket.Packet, counters fwdobject.C
 			key = append(key, f...)
 		}
 	}
-	index := s.hashFn(key)
-	if index >= len(s.set) {
-		index -= len(s.set) * (index / len(s.set))
+
+	h := s.hashFn(key, s.weightSum)
```

Comment:
Why can't mod (%) be used here to avoid changing the hash functions?

Reply:
changed